### ISSUE TITLE
json_decode()/js_decode() value is locked automatically

### ISSUE THREAD

**Issue by Shougo** [vim/vim#673]
``` vim
let plugins = {'foo': {'sourced': 0}}
let plugins2 = json_decode(json_encode(plugins))
let plugins2['foo'].sourced = 1
```

I get the error message.

`E741: Value is locked: plugins2['foo'].sourced = 1`

Why the value is locked?

``` vim
let plugins = {'foo': {'sourced': 0}}
let plugins2 = js_decode(js_encode(plugins))
let plugins2['foo'].sourced = 1
```

The same problem exists for `js_decode()`.

**Comment by brammool** [comment]
Shougo wrote:

> ``` vim
> let plugins = {'foo': {'sourced': 0}}
> let plugins2 = json_decode(json_encode(plugins))
> let plugins2['foo'].sourced = 1
> ```
> 
> I get the error message.
> 
> `E741: Value is locked: plugins2['foo'].sourced = 1`
> 
> Why the value is locked?
> 
> ``` vim
> let plugins = {'foo': {'sourced': 0}}
> let plugins2 = js_decode(js_encode(plugins))
> let plugins2['foo'].sourced = 1
> ```
> 
> The same problem exists for `js_decode()`.

I cannot reproduce the error.  Looks like something is not initialized.
Ah, valgrind shows it.

## 

Engineers are always delighted to share wisdom, even in areas in which they
have no experience whatsoever.  Their logic provides them with inherent
insight into any field of expertise.  This can be a problem when dealing with
the illogical people who believe that knowledge can only be derived through
experience.
                (Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by Shougo** [comment]
The problem is fixed in Vim 7.4.1498.  Thank you!
